lib/autofocus.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * When triggered, automatically focus the first element ref passed to this object.
+ *
+ * To unconditionally focus a single element:
+ *
+ * ```
+ * class SomeComponent extends React.Component {
+ *   constructor(props) {
+ *     super(props);
+ *     this.autofocus = new Autofocus();
+ *   }
+ *
+ *   render() {
+ *     return (
+ *       <div className="github-Form">
+ *         <input ref={this.autofocus.target} type="text" />
+ *         <input type="text" />
+ *       </div>
+ *     );
+ *   }
+ *
+ *   componentDidMount() {
+ *     this.autofocus.trigger();
+ *   }
+ * }
+ * ```
+ *
+ * If multiple form elements are present, use `firstTarget` to create the ref instead. The rendered ref you assign the
+ * lowest numeric index will be focused on trigger:
+ *
+ * ```
+ * class SomeComponent extends React.Component {
+ *   constructor(props) {
+ *     super(props);
+ *     this.autofocus = new Autofocus();
+ *   }
+ *
+ *   render() {
+ *     return (
+ *       <div className="github-Form">
+ *         {this.props.someProp && <input ref={this.autofocus.firstTarget(0)} />}
+ *         <input ref={this.autofocus.firstTarget(1)} type="text" />
+ *         <input type="text" />
+ *       </div>
+ *     );
+ *   }
+ *
+ *   componentDidMount() {
+ *     this.autofocus.trigger();
+ *   }
+ * }
+ * ```
+ *
+ */
+export default class AutoFocus {
```

Comment:
I'm wondering if I could generalize this somehow to deal with #1403. It'd likely need to wait on a hooks refactor, though, and be passed as a context... 🤔